### PR TITLE
tests: boards: nrf: qdec: add tests for qdec for nrf

### PR DESCRIPTION
**Pull request by nordic-piks**
Expand testing for QDEC at nrf platforms.
It uses general sensor API,
however there are also nrf driver specific assumptions.

### CODE REVIEW

**Review by nordic-segl — APPROVED** [review]

**Review by jaz1-nordic — DISMISSED** [review]

**Review by nordic-segl — APPROVED** [review]

**Review by kl-cruz — DISMISSED** [review]

File: tests/boards/nrf/qdec/src/main.c [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+
+K_SEM_DEFINE(sem, 0, 1);
```
Comment (anangl) [inline tests/boards/nrf/qdec/src/main.c:14]:
```suggestion
static K_SEM_DEFINE(sem, 0, 1);
```

File: tests/boards/nrf/qdec/src/main.c [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+	zassert_equal_ptr(dev, qdec_dev);
+	/* trigger should be stored as a pointer in a driver
+	 * thus this address should match
+	 */
+	zassert_equal_ptr(trigger, &qdec_trigger);
+
+	k_sem_give(&sem);
+}
+
+void qenc_emulate_work_handler(struct k_work *work)
```
Comment (anangl) [inline tests/boards/nrf/qdec/src/main.c:36]:
```suggestion
static void qenc_emulate_work_handler(struct k_work *work)
```

File: tests/boards/nrf/qdec/src/main.c [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	/* wait for some readings*/
+	k_msleep(emulation_duration_ms);
+
+	rc = sensor_sample_fetch(qdec_dev);
+	zassert_true(rc == 0, "Failed to fetch sample (%d)", rc);
+
+	rc = sensor_channel_get(qdec_dev, SENSOR_CHAN_ROTATION, &val);
+	zassert_true(rc == 0, "Failed to get sample (%d)", rc);
+
+	printk("QDEC reading: %d\n", val.val1);
```
Comment (anangl) [inline tests/boards/nrf/qdec/src/main.c:116]: Why not `TC_PRINT()`, actually?

File: tests/boards/nrf/qdec/src/main.c (hunk 1 above)
Comment (nordic-piks) [reply tests/boards/nrf/qdec/src/main.c]: Changed, thanks :)

File: tests/boards/nrf/qdec/src/main.c (hunk 2 above)
Comment (nordic-piks) [reply tests/boards/nrf/qdec/src/main.c]: Changed, thanks :)

File: tests/boards/nrf/qdec/src/main.c (hunk 3 above)
Comment (nordic-piks) [reply tests/boards/nrf/qdec/src/main.c]: Changed, thanks ;)

Comment (anangl) [reply tests/boards/nrf/qdec/src/main.c]: Why only this one occurrence? :)

**Review by anangl — DISMISSED** [review]

File: tests/boards/nrf/qdec/src/main.c (hunk 3 above)
Comment (nordic-piks) [reply tests/boards/nrf/qdec/src/main.c]: Sorry, now fixed all occurrences.

**Review by anangl — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Pull request merged**